### PR TITLE
[REF] web, website: remove legacy DebugManager and use new DebugMenu

### PR DESCRIPTION
**Pull request by sdegueldre**
Previously, the debug manager was rewritten with the new webclient
architecture, however, because the frontend did not have access to the
new infrastructure (env and services), the old debug manager was kept
and used only in the frontend.

Since odoo/odoo#72675 the new env and services are now available, this
commit simply uses the new debug manager in the frontend and removes the
legacy code.

### CODE REVIEW

File: addons/web/static/src/core/debug/debug_menu_items.js [hunk 1]
```diff
@@ -158,7 +114,6 @@ registry
     // Backend
     .add("runJSTestsItem", runJSTestsItem)
     .add("runJSTestsMobileItem", runJSTestsMobileItem)
```
Comment (sdegueldre) [inline addons/web/static/src/core/debug/debug_menu_items.js:116]: These are now available also in the frontend, so maybe update comment? Or do we *not* want them in the frontend?

File: addons/web/static/src/core/debug/profiling/profiling_item.js [hunk 2]
```diff
@@ -8,7 +8,9 @@ const { Component } = owl;
 export class ProfilingItem extends Component {
     setup() {
         this.profiling = useService("profiling");
-        this.actionService = useService("action");
+        if (this.env.services.action) {
+            this.actionService = useService("action");
+        }
```
Comment (sdegueldre) [inline addons/web/static/src/core/debug/profiling/profiling_item.js:13]: use a link to the xmlid of the action instead

File: addons/web/static/src/legacy/js/public/public_root.js [hunk 3]
```diff
@@ -377,8 +378,10 @@ export async function createPublicRoot(RootWidget) {
     await Promise.all([owl.utils.whenReady(), session.is_bound]);
 
     const wowlEnv = makeEnv();
+    window.wowlEnv = wowlEnv;
```
Comment (sdegueldre) [inline addons/web/static/src/legacy/js/public/public_root.js:381]: Yikes. Find better way to get wowlEnv when spawning debug menu.

File: addons/web/static/src/core/debug/debug_menu_items.js (hunk 1 above)
Comment (aab-odoo) [reply addons/web/static/src/core/debug/debug_menu_items.js:116]: in master, they aren't in the frontend

**Review by aab-odoo — APPROVED** [review]
seems good already

File: addons/web/static/src/core/debug/debug_menu_items.js (hunk 1 above)
Comment (sdegueldre) [reply addons/web/static/src/core/debug/debug_menu_items.js]: I checked with the website team, they do not want those items in the frontend so I have moved the definition of these items to the webclient.

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**